### ISSUE TITLE
Enhancements to docs, scripts, tests for new contributors

### ISSUE THREAD

**Issue by humphd** [Seneca-CDOT/plumadriver#117]
**What would you like to be added**:

I spoke with @ctyler about getting some of my students involved in working on this project. In order to accelerate this, I started to work on a fix myself.  Right away I ran into issues with docs and npm scripts.

First, the docs for building reference incorrect scripts.

```diff
diff --git a/README.md b/README.md
index 8dd0e12..da9c474 100644
--- a/README.md
+++ b/README.md
@@ -105,7 +105,6 @@ pluma.get("http://www.example.com");

 ### Requirements:
  - Latest stable version of NodeJS
- - npm pkg  module `npm install pkg -g`

 From the command line:

@@ -113,10 +112,10 @@ From the command line:
 2. `cd plumadriver`
 3. `npm install`
 4. For Linux:
-    `npm run build-linux`
+    `npm run build:linux`
    For Windows:
-    `npm run build-win`
+    `npm run build:win`
    For Mac:
-    `npm run build-macos`
+    `npm run build:macos`

 These scripts will transpile all the typescript code found within src directory and output the result to the build directory. From here pkg will create an executable which should appear in the project's root directory.
\ No newline at end of file
~
```

The note to install `pkg` globally isn't needed any more, since it's been included in `package.json` as a dev despondency.

I then tried to update these docs:

```
$ git commit -m "Update build docs to reflect current state of master"
husky > pre-commit (node v12.13.1)
  ↓ Stashing changes... [skipped]
    → No partially staged files found...
  ❯ Running tasks...
    ↓ Running tasks for *.{json,css,scss,html} [skipped]
      → No staged files match *.{json,css,scss,html}
    ❯ Running tasks for *.md
      ✔ prettier --write
      ✖ markdownlint
        git add
    ↓ Running tasks for *.{js,ts} [skipped]
      → No staged files match *.{js,ts}

✖ markdownlint found some errors. Please fix them and try committing again.
husky > pre-commit hook failed (add --no-verify to bypass)
```

I'm unable to commit due to markdown linting; however, I get zero information about how to fix it.  What are the errors?  Which lines?  There's no way to tell, and therefore, no way to fix.

Looking at [.huskyrc.json](.huskyrc.json) I can see that it's calling `prettier` and `markdownlint` for `*.md` files.  However, there are no equivalent user-facing scripts for running these in `package.json`.

Why is `prettier` generating Markdown that `markdownlint` won't accept?  Why not just take the output of `prettier`?

If I remove the `markdownlint` step, I run into this:

```
$ git commit -m "Update build docs to reflect current state of master"
husky > pre-commit (node v12.13.1)
  ✔ Stashing changes...
  ✔ Running tasks...
  ✔ Updating stash...
  ✔ Restoring local changes...
husky > commit-msg (node v12.13.1)
⧗   input: Update build docs to reflect current state of master
✖   subject may not be empty [subject-empty]
✖   type may not be empty [type-empty]

✖   found 2 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

husky > commit-msg hook failed (add --no-verify to bypass)
```

There are no indications in the docs about how commit messages should be formatted.  If you're going to require strict commit message formatting, why not include a tool like [cz](https://github.com/commitizen/cz-cli#if-your-repo-is-commitizen-friendly) to help devs generate appropriate messages?

At this stage, many contributors will give up.  I'm willing to keep going, but I wanted to flag this for you, as it's a lot more frustrating than it needs to be.

I also note that you're failing 7 tests when I run `npm test`.  I don't see any issues filed on this, so it's unclear if they are being dealt with in other issues, or if anyone is aware of the problem.

**Why is this needed**:

Trying to contribute to the project at the moment is more frustrating that it needs to be.  It should be easy to get started working on the code as an external contributor.

I wanted to make you aware of all this, since it's likely not something you run into on a daily basis, because you "know how everything works."  New devs will have no idea, and the more automatic you can make the process, and the more "guard rails" you can put in place, the better the experience will be for them and you.

**Comment by poftadeh** [comment]
Thank you for the detailed feedback. I'm aware that the repo is not very accommodating to new contributors, and work is on the way to address this as soon as possible.  The `readme` needs updating as you've pointed out, since it does not reflect the changes made since previous semesters' work. No longer needing to install `pkg` globally is one such change. The list of endpoints, and instructions on using PlumaDriver will also need some updating. 

> Why is prettier generating Markdown that markdownlint won't accept?

This appears to be a bug in the configuration for handling this type of file. `markdownlint` should be removed from the `lintstaged` config. I will look into this, thanks.

> I also note that you're failing 7 tests when I run npm test. I don't see any issues filed on this, so it's unclear if they are being dealt with in other issues, or if anyone is aware of the problem.

I've been adding tests recently to endpoints worked on in previous semesters that I noticed were missing coverage. This exposed several bugs in the older parts of the code (with more likely to come as I further increase coverage). I've opened individual issues for each one of these newly discovered bugs. However, another issue or note acknowledging the fact that the tests are failing in general would be prudent, and would make things clearer for new contributors.

> There are no indications in the docs about how commit messages should be formatted. If you're going to require strict commit message formatting, why not include a tool like cz to help devs generate appropriate messages?

`commitlint` was added to enforce some internal consistency on our end, but I will likely remove it now that the project is opening up to the community. In the meantime, feel free to bypass husky with `--no-verify`.

**Comment by humphd** [comment]
I also note that manual linting doesn't work:

```
$ npm run lint

> plumadriver@0.5.0 lint /Users/humphd/repos/plumadriver
> eslint

eslint [options] file.js [file.js] [dir]

Basic configuration:
  --no-eslintrc                   Disable use of configuration from .eslintrc.*
  -c, --config path::String       Use this configuration, overriding
                                  .eslintrc.* config options if present
  --env [String]                  Specify environments
  --ext [String]                  Specify JavaScript file extensions -
                                  default: .js
  --global [String]               Define global variables
  --parser String                 Specify the parser to be used
  --parser-options Object         Specify parser options
  --resolve-plugins-relative-to path::String  A folder where plugins should be
                                              resolved from, CWD by default

Specifying rules and plugins:
  --rulesdir [path::String]       Use additional rules from this directory
  --plugin [String]               Specify plugins
  --rule Object                   Specify rules

Fixing problems:
  --fix                           Automatically fix problems
  --fix-dry-run                   Automatically fix problems without saving
                                  the changes to the file system
  --fix-type Array                Specify the types of fixes to apply
                                  (problem, suggestion, layout)

Ignoring files:
  --ignore-path path::String      Specify path of ignore file
  --no-ignore                     Disable use of ignore files and patterns
  --ignore-pattern [String]       Pattern of files to ignore (in addition to
                                  those in .eslintignore)

Using stdin:
  --stdin                         Lint code provided on <STDIN> - default:
                                  false
  --stdin-filename String         Specify filename to process STDIN as

Handling warnings:
  --quiet                         Report errors only - default: false
  --max-warnings Int              Number of warnings to trigger nonzero exit
                                  code - default: -1

Output:
  -o, --output-file path::String  Specify file to write report to
  -f, --format String             Use a specific output format - default:
                                  stylish
  --color, --no-color             Force enabling/disabling of color

Inline configuration comments:
  --no-inline-config              Prevent comments from changing config or
                                  rules
  --report-unused-disable-directives  Adds reported errors for unused
                                      eslint-disable directives

Caching:
  --cache                         Only check changed files - default: false
  --cache-file path::String       Path to the cache file. Deprecated: use
                                  --cache-location - default: .eslintcache
  --cache-location path::String   Path to the cache file or directory

Miscellaneous:
  --init                          Run config initialization wizard - default:
                                  false
  --debug                         Output debugging information
  -h, --help                      Show help
  -v, --version                   Output the version number
  --print-config path::String     Print the configuration for the given file
```

**Comment by poftadeh** [comment]
@humphd 

I've made a few updates to the repo:

- the readme has been changed fixing broken links, adding new content, and updating instructions. A new section has been added for those who wish to try out PlumaDriver without Selenium.

- fixed `markdownlint` bug caused when attempting to commit changes to markdown files.

- fixed npm lint scripts.

- added a section to `CONTRIBUTING.md` documenting the commit message format. A new script `npm run commit` was added for interactive commits using commitizen.

Hopefully this will help out newcomers to the repo.
